hazelcast/src/main/java/com/hazelcast/sql/SqlRowMetadata.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql;
+
+import com.hazelcast.internal.util.Preconditions;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * SQL row metadata.
+ */
+public class SqlRowMetadata {
+    /** Constant indicating that the column is not found. */
+    public static final int COLUMN_NOT_FOUND = -1;
+
+    private final List<SqlColumnMetadata> columns;
+    private final Map<String, Integer> nameToIndex;
+
+    public SqlRowMetadata(List<SqlColumnMetadata> columns) {
+        assert columns != null && !columns.isEmpty();
+
+        this.columns = new ArrayList<>(columns);
+
+        nameToIndex = new HashMap<>(columns.size());
+
+        for (int i = 0; i < columns.size(); i++) {
+            nameToIndex.put(columns.get(i).getName(), i);
+        }
+    }
+
+    /**
+     * Gets the number of columns in the row.
+     *
+     * @return The number of columns in the row.
+     */
+    public int getColumnCount() {
+        return columns.size();
+    }
+
+    /**
+     * Get column metadata.
+     *
+     * @param index Column index, 0-based.
```

Comment:
veery minor but typically the javadoc for `@param` statements start in lower case everywhere else

Reply:
Applied to `@param`, `@return`, `@throws`.